scripts/cibuild/SetVstsBuildNumber.ps1

```diff
@@ -0,0 +1,41 @@
+# This script is used to dynamically set some build variables during VSTS build.
+# Specifically, this script reads the buildcounter.txt file in the $(DropRoot) to
+# determine the build number of the artifacts, also it sets the $(NupkgOutputDir)
+# based on whether $(BuildRTM) is true or false.
+param
+(
+    [string]$BuildCounter,
+    [string]$BuildInfoJson,
+    [string]$BuildRTM
+)
+
+if ($BuildRTM -eq 'true')
+{
+    Write-Host "##vso[task.setvariable variable=NupkgOutputDir;]ReleaseNupkgs"
+    do{
+        Start-Sleep -s 15
+        Write-Host "Waiting for buildinfo.json to be generated..."
+    }
+    until (Test-Path $BuildInfoJson)
+    $json = (Get-Content $BuildInfoJson -Raw) | ConvertFrom-Json
+    $currentBuild = [System.Decimal]::Parse($json.BuildNumber)
+    Write-Host "##vso[task.setvariable variable=Revision;]$currentBuild" 
+}
+else
+{
+    $revision = Get-Content $BuildCounter
+    $newBuildCounter = [System.Decimal]::Parse($revision)
+    $newBuildCounter++
+    Set-Content $BuildCounter $newBuildCounter
+    Write-Host "##vso[task.setvariable variable=Revision;]$newBuildCounter"
+    $jsonRepresentation = "
+    {
+        `"BuildNumber`":  `"$newBuildCounter`",
```

Comment:
@alpaix this was actually a path to a file, so i have updated the variable names accordingly